public/app/features/scenes/dashboard/DashboardsLoader.ts

```diff
@@ -136,6 +159,76 @@ export class DashboardLoader extends StateManagerBase<DashboardLoaderState> {
 
     return panels;
   }
+
+  private migrateVariables(dashboard: DashboardModel) {
+    return (
+      dashboard.templating.list
+        .map((v) => {
+          try {
+            return createVariableFromLegacyModel(v);
+          } catch (err) {
+            console.error(err);
+            return null;
+          }
+        })
+        // TODO: Remove filter
+        // Added temporarily to allow skipping non-compatible variables
+        .filter((v): v is SceneVariable => Boolean(v))
+    );
+  }
+}
+
+export function createVariableFromLegacyModel(variable: VariableModel): SceneVariable {
```

Comment:
If I'm not wrong, we also support [ConstantVariable](https://github.com/grafana/grafana/blob/main/public/app/features/scenes/variables/variants/ConstantVariable.ts).

Reply:
Pushed a change with that just now